### ISSUE TITLE
Can not deploy to heroku

### ISSUE THREAD

**Issue by speamy** [hackmdio/hackmd#814]
Hi, 

I having problems to running hackmd on heroku. I use hackmd over the Button in the readme.
I get this backtrace:
```
-----> VIPS app detected
-----> Installing libvips
       Downloading libvips and unpacking
       libvips installed
-----> Node.js app detected
-----> Creating runtime environment
       
       NPM_CONFIG_LOGLEVEL=error
       NPM_CONFIG_PRODUCTION=false
       NODE_VERBOSE=false
       NODE_ENV=production
       NODE_MODULES_CACHE=true
-----> Installing binaries
       engines.node (package.json):  >=6.x
       engines.npm (package.json):   unspecified (use default)
       engines.yarn (package.json):  unspecified (use default)
       
       Resolving node version >=6.x...
       Downloading and installing node 10.0.0...
       Using default npm version: 5.6.0
       Resolving yarn version 1.x...
       Downloading and installing yarn (1.6.0)...
       Installed yarn 1.6.0
-----> Restoring cache
       Skipping cache restore (not-found)
-----> Building dependencies
       Installing node modules (yarn.lock)
       yarn install v1.6.0
       (node:287) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
       [1/4] Resolving packages...
       [2/4] Fetching packages...
       warning Pattern ["raphael@git+https://github.com/dmitrybaranovskiy/raphael"] is trying to unpack in the same destination "/tmp/yarncache.Sd0bX/v1/npm-raphael-2.2.7-fe8e591e1c86b5aeb4c252b33c08e647434504c5" as pattern ["raphael@2.2.7"]. This could result in non-deterministic behavior, skipping.
       info No lockfile found.
       [1/4] Resolving packages...
       warning blint > nomnom@1.8.1: Package no longer supported. Contact support@npmjs.com for more info.
       [2/4] Fetching packages...
       [3/4] Linking dependencies...
       [4/4] Building fresh packages...
       success Saved lockfile.
       $ npm run-script build && ./release.sh
       
       > codemirror@5.29.1 build /tmp/yarncache.Sd0bX/v1/.tmp/494d4cd821398a8aa6f314056acc7b33.c4225222bbb7cf7d3fbcfeeb4b9de0a644b2537e.prepare
       > rollup -c
       
       WARN: Non-strict equality against boolean: == false [lib/codemirror.js:1300,59]
       WARN: Non-strict equality against boolean: != false [addon/comment/comment.js:107,55]
       WARN: Non-strict equality against boolean: != false [addon/comment/comment.js:120,10]
       WARN: Non-strict equality against boolean: == false [mode/javascript/javascript.js:432,13]
       WARN: Non-strict equality against boolean: == false [mode/javascript/javascript.js:433,15]
       WARN: Non-strict equality against boolean: != false [mode/javascript/javascript.js:799,55]
       $ npm run-script build
       
       > codemirror@5.29.1 build /tmp/yarncache.Sd0bX/v1/.tmp/494d4cd821398a8aa6f314056acc7b33.c4225222bbb7cf7d3fbcfeeb4b9de0a644b2537e.prepare
       > rollup -c
       
       info fsevents@1.1.2: The platform "linux" is incompatible with this module.
       info "fsevents@1.1.2" is an optional dependency and failed compatibility check. Excluding it from installation.
       [3/4] Linking dependencies...
       warning "js-url > grunt-contrib-jshint@1.1.0" has unmet peer dependency "grunt@>=0.4.0".
       [4/4] Building fresh packages...
       error /tmp/build_431d75b25ac1fa5430285940870b0ae7/node_modules/scrypt: Command failed.
       Exit code: 1
       Command: node-gyp rebuild
       Arguments:
       Directory: /tmp/build_431d75b25ac1fa5430285940870b0ae7/node_modules/scrypt
       Output:
       make: Entering directory '/tmp/build_431d75b25ac1fa5430285940870b0ae7/node_modules/scrypt/build'
       SOLINK_MODULE(target) Release/obj.target/copied_files.node
       COPY Release/copied_files.node
       CC(target) Release/obj.target/scrypt_wrapper/src/util/memlimit.o
       CC(target) Release/obj.target/scrypt_wrapper/src/scryptwrapper/keyderivation.o
       CC(target) Release/obj.target/scrypt_wrapper/src/scryptwrapper/pickparams.o
       CC(target) Release/obj.target/scrypt_wrapper/src/scryptwrapper/hash.o
       AR(target) Release/obj.target/scrypt_wrapper.a
       COPY Release/scrypt_wrapper.a
       CC(target) Release/obj.target/scrypt_lib/scrypt/scrypt-1.2.0/lib/crypto/crypto_scrypt.o
       CC(target) Release/obj.target/scrypt_lib/scrypt/scrypt-1.2.0/lib/crypto/crypto_scrypt_smix.o
       CC(target) Release/obj.target/scrypt_lib/scrypt/scrypt-1.2.0/libcperciva/util/warnp.o
       CC(target) Release/obj.target/scrypt_lib/scrypt/scrypt-1.2.0/libcperciva/alg/sha256.o
       CC(target) Release/obj.target/scrypt_lib/scrypt/scrypt-1.2.0/libcperciva/util/insecure_memzero.o
       CC(target) Release/obj.target/scrypt_lib/scrypt/scrypt-1.2.0/lib/scryptenc/scryptenc_cpuperf.o
       AR(target) Release/obj.target/scrypt_lib.a
       COPY Release/scrypt_lib.a
       CXX(target) Release/obj.target/scrypt/src/node-boilerplate/scrypt_common.o
       In file included from ../../nan/nan.h:192:0,
       from ../src/node-boilerplate/scrypt_common.cc:28:
       ../../nan/nan_maybe_43_inl.h: In function ‘Nan::Maybe<bool> Nan::ForceSet(v8::Local<v8::Object>, v8::Local<v8::Value>, v8::Local<v8::Value>, v8::PropertyAttribute)’:
       ../../nan/nan_maybe_43_inl.h:112:15: error: ‘class v8::Object’ has no member named ‘ForceSet’
       return obj->ForceSet(isolate->GetCurrentContext(), key, value, attribs);
       ^
       In file included from ../src/node-boilerplate/scrypt_common.cc:28:0:
       ../../nan/nan.h: In function ‘v8::Local<v8::Value> Nan::MakeCallback(v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*)’:
       ../../nan/nan.h:835:60: warning: ‘v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*)’ is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
       v8::Isolate::GetCurrent(), target, func, argc, argv);
       ^
       In file included from ../../nan/nan.h:49:0,
       from ../src/node-boilerplate/scrypt_common.cc:28:
       /app/.node-gyp/10.0.0/include/node/node.h:172:50: note: declared here
       NODE_EXTERN v8::Local<v8::Value> MakeCallback(
       ^
       /app/.node-gyp/10.0.0/include/node/node.h:88:42: note: in definition of macro ‘NODE_DEPRECATED’
       __attribute__((deprecated(message))) declarator
       ^
       In file included from ../src/node-boilerplate/scrypt_common.cc:28:0:
       ../../nan/nan.h: In function ‘v8::Local<v8::Value> Nan::MakeCallback(v8::Local<v8::Object>, v8::Local<v8::String>, int, v8::Local<v8::Value>*)’:
       ../../nan/nan.h:850:62: warning: ‘v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::String>, int, v8::Local<v8::Value>*)’ is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
       v8::Isolate::GetCurrent(), target, symbol, argc, argv);
       ^
       In file included from ../../nan/nan.h:49:0,
       from ../src/node-boilerplate/scrypt_common.cc:28:
       /app/.node-gyp/10.0.0/include/node/node.h:165:50: note: declared here
       NODE_EXTERN v8::Local<v8::Value> MakeCallback(
       ^
       /app/.node-gyp/10.0.0/include/node/node.h:88:42: note: in definition of macro ‘NODE_DEPRECATED’
       __attribute__((deprecated(message))) declarator
       ^
       In file included from ../src/node-boilerplate/scrypt_common.cc:28:0:
       ../../nan/nan.h: In function ‘v8::Local<v8::Value> Nan::MakeCallback(v8::Local<v8::Object>, const char*, int, v8::Local<v8::Value>*)’:
       ../../nan/nan.h:865:62: warning: ‘v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, const char*, int, v8::Local<v8::Value>*)’ is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
       v8::Isolate::GetCurrent(), target, method, argc, argv);
       ^
       In file included from ../../nan/nan.h:49:0,
       from ../src/node-boilerplate/scrypt_common.cc:28:
       /app/.node-gyp/10.0.0/include/node/node.h:158:50: note: declared here
       NODE_EXTERN v8::Local<v8::Value> MakeCallback(
       ^
       /app/.node-gyp/10.0.0/include/node/node.h:88:42: note: in definition of macro ‘NODE_DEPRECATED’
       __attribute__((deprecated(message))) declarator
       ^
       In file included from ../src/node-boilerplate/scrypt_common.cc:28:0:
       ../../nan/nan.h: In member function ‘v8::Local<v8::Value> Nan::Callback::Call_(v8::Isolate*, v8::Local<v8::Object>, int, v8::Local<v8::Value>*) const’:
       ../../nan/nan.h:1479:5: warning: ‘v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*)’ is deprecated: Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
       ));
       ^
       In file included from ../../nan/nan.h:49:0,
       from ../src/node-boilerplate/scrypt_common.cc:28:
       /app/.node-gyp/10.0.0/include/node/node.h:172:50: note: declared here
       NODE_EXTERN v8::Local<v8::Value> MakeCallback(
       ^
       /app/.node-gyp/10.0.0/include/node/node.h:88:42: note: in definition of macro ‘NODE_DEPRECATED’
       __attribute__((deprecated(message))) declarator
       ^
       scrypt.target.mk:115: recipe for target 'Release/obj.target/scrypt/src/node-boilerplate/scrypt_common.o' failed
       make: Leaving directory '/tmp/build_431d75b25ac1fa5430285940870b0ae7/node_modules/scrypt/build'
       make: *** [Release/obj.target/scrypt/src/node-boilerplate/scrypt_common.o] Error 1
       gyp ERR! build error
       gyp ERR! stack Error: `make` failed with exit code: 2
       gyp ERR! stack     at ChildProcess.onExit (/tmp/build_431d75b25ac1fa5430285940870b0ae7/.heroku/node/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:258:23)
       gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
       gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:225:12)
       gyp ERR! System Linux 4.4.0-1014-aws
       gyp ERR! command "/tmp/build_431d75b25ac1fa5430285940870b0ae7/.heroku/node/bin/node" "/tmp/build_431d75b25ac1fa5430285940870b0ae7/.heroku/node/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
       gyp ERR! cwd /tmp/build_431d75b25ac1fa5430285940870b0ae7/node_modules/scrypt
       gyp ERR! node -v v10.0.0
       gyp ERR! node-gyp -v v3.6.2
       gyp ERR! not ok
       info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
-----> Build failed
       
       We're sorry this build is failing! You can troubleshoot common issues here:
       https://devcenter.heroku.com/articles/troubleshooting-node-deploys
       
       Some possible problems:
       
       - Dangerous semver range (>) in engines.node
       https://devcenter.heroku.com/articles/nodejs-support#specifying-a-node-js-version
       
       Love,
       Heroku
       
 !     Push rejected, failed to compile Node.js app.
 !     Push failed

```

**Comment by SISheogorath** [comment]
Yes, I already noticed this. It's cause vy scrypt incompatibility with node version 10.

So basically it's an upstream problem. Haven't looked into the details yet but I hope it can be resolved without maintaining an own fork.

On 6 May 2018 18:59:58 GMT+02:00, speamy <notifications@github.com> wrote:
>Hi, 
>
>I have been having problems running hackmd on heroku. I use hackmd over
>the Button in the readme.
>I become this backstrace:
>```
>-----> VIPS app detected
>-----> Installing libvips
>       Downloading libvips and unpacking
>       libvips installed
>-----> Node.js app detected
>-----> Creating runtime environment
>       
>       NPM_CONFIG_LOGLEVEL=error
>       NPM_CONFIG_PRODUCTION=false
>       NODE_VERBOSE=false
>       NODE_ENV=production
>       NODE_MODULES_CACHE=true
>-----> Installing binaries
>       engines.node (package.json):  >=6.x
>       engines.npm (package.json):   unspecified (use default)
>       engines.yarn (package.json):  unspecified (use default)
>       
>       Resolving node version >=6.x...
>       Downloading and installing node 10.0.0...
>       Using default npm version: 5.6.0
>       Resolving yarn version 1.x...
>       Downloading and installing yarn (1.6.0)...
>       Installed yarn 1.6.0
>-----> Restoring cache
>       Skipping cache restore (not-found)
>-----> Building dependencies
>       Installing node modules (yarn.lock)
>       yarn install v1.6.0
>(node:287) [DEP0005] DeprecationWarning: Buffer() is deprecated due to
>security and usability issues. Please use the Buffer.alloc(),
>Buffer.allocUnsafe(), or Buffer.from() methods instead.
>       [1/4] Resolving packages...
>       [2/4] Fetching packages...
>warning Pattern
>["raphael@git+https://github.com/dmitrybaranovskiy/raphael"] is trying
>to unpack in the same destination
>"/tmp/yarncache.Sd0bX/v1/npm-raphael-2.2.7-fe8e591e1c86b5aeb4c252b33c08e647434504c5"
>as pattern ["raphael@2.2.7"]. This could result in non-deterministic
>behavior, skipping.
>       info No lockfile found.
>       [1/4] Resolving packages...
>warning blint > nomnom@1.8.1: Package no longer supported. Contact
>support@npmjs.com for more info.
>       [2/4] Fetching packages...
>       [3/4] Linking dependencies...
>       [4/4] Building fresh packages...
>       success Saved lockfile.
>       $ npm run-script build && ./release.sh
>       
>> codemirror@5.29.1 build
>/tmp/yarncache.Sd0bX/v1/.tmp/494d4cd821398a8aa6f314056acc7b33.c4225222bbb7cf7d3fbcfeeb4b9de0a644b2537e.prepare
>       > rollup -c
>       
>WARN: Non-strict equality against boolean: == false
>[lib/codemirror.js:1300,59]
>WARN: Non-strict equality against boolean: != false
>[addon/comment/comment.js:107,55]
>WARN: Non-strict equality against boolean: != false
>[addon/comment/comment.js:120,10]
>WARN: Non-strict equality against boolean: == false
>[mode/javascript/javascript.js:432,13]
>WARN: Non-strict equality against boolean: == false
>[mode/javascript/javascript.js:433,15]
>WARN: Non-strict equality against boolean: != false
>[mode/javascript/javascript.js:799,55]
>       $ npm run-script build
>       
>> codemirror@5.29.1 build
>/tmp/yarncache.Sd0bX/v1/.tmp/494d4cd821398a8aa6f314056acc7b33.c4225222bbb7cf7d3fbcfeeb4b9de0a644b2537e.prepare
>       > rollup -c
>       
>info fsevents@1.1.2: The platform "linux" is incompatible with this
>module.
>info "fsevents@1.1.2" is an optional dependency and failed
>compatibility check. Excluding it from installation.
>       [3/4] Linking dependencies...
>warning "js-url > grunt-contrib-jshint@1.1.0" has unmet peer dependency
>"grunt@>=0.4.0".
>       [4/4] Building fresh packages...
>error /tmp/build_431d75b25ac1fa5430285940870b0ae7/node_modules/scrypt:
>Command failed.
>       Exit code: 1
>       Command: node-gyp rebuild
>       Arguments:
>Directory:
>/tmp/build_431d75b25ac1fa5430285940870b0ae7/node_modules/scrypt
>       Output:
>make: Entering directory
>'/tmp/build_431d75b25ac1fa5430285940870b0ae7/node_modules/scrypt/build'
>       SOLINK_MODULE(target) Release/obj.target/copied_files.node
>       COPY Release/copied_files.node
>       CC(target) Release/obj.target/scrypt_wrapper/src/util/memlimit.o
>CC(target)
>Release/obj.target/scrypt_wrapper/src/scryptwrapper/keyderivation.o
>CC(target)
>Release/obj.target/scrypt_wrapper/src/scryptwrapper/pickparams.o
>  CC(target) Release/obj.target/scrypt_wrapper/src/scryptwrapper/hash.o
>       AR(target) Release/obj.target/scrypt_wrapper.a
>       COPY Release/scrypt_wrapper.a
>CC(target)
>Release/obj.target/scrypt_lib/scrypt/scrypt-1.2.0/lib/crypto/crypto_scrypt.o
>CC(target)
>Release/obj.target/scrypt_lib/scrypt/scrypt-1.2.0/lib/crypto/crypto_scrypt_smix.o
>CC(target)
>Release/obj.target/scrypt_lib/scrypt/scrypt-1.2.0/libcperciva/util/warnp.o
>CC(target)
>Release/obj.target/scrypt_lib/scrypt/scrypt-1.2.0/libcperciva/alg/sha256.o
>CC(target)
>Release/obj.target/scrypt_lib/scrypt/scrypt-1.2.0/libcperciva/util/insecure_memzero.o
>CC(target)
>Release/obj.target/scrypt_lib/scrypt/scrypt-1.2.0/lib/scryptenc/scryptenc_cpuperf.o
>       AR(target) Release/obj.target/scrypt_lib.a
>       COPY Release/scrypt_lib.a
>CXX(target)
>Release/obj.target/scrypt/src/node-boilerplate/scrypt_common.o
>       In file included from ../../nan/nan.h:192:0,
>       from ../src/node-boilerplate/scrypt_common.cc:28:
>../../nan/nan_maybe_43_inl.h: In function ‘Nan::Maybe<bool>
>Nan::ForceSet(v8::Local<v8::Object>, v8::Local<v8::Value>,
>v8::Local<v8::Value>, v8::PropertyAttribute)’:
>../../nan/nan_maybe_43_inl.h:112:15: error: ‘class v8::Object’ has no
>member named ‘ForceSet’
>return obj->ForceSet(isolate->GetCurrentContext(), key, value,
>attribs);
>       ^
>   In file included from ../src/node-boilerplate/scrypt_common.cc:28:0:
>../../nan/nan.h: In function ‘v8::Local<v8::Value>
>Nan::MakeCallback(v8::Local<v8::Object>, v8::Local<v8::Function>, int,
>v8::Local<v8::Value>*)’:
>../../nan/nan.h:835:60: warning: ‘v8::Local<v8::Value>
>node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>,
>v8::Local<v8::Function>, int, v8::Local<v8::Value>*)’ is deprecated:
>Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
>       v8::Isolate::GetCurrent(), target, func, argc, argv);
>       ^
>       In file included from ../../nan/nan.h:49:0,
>       from ../src/node-boilerplate/scrypt_common.cc:28:
>  /app/.node-gyp/10.0.0/include/node/node.h:172:50: note: declared here
>       NODE_EXTERN v8::Local<v8::Value> MakeCallback(
>       ^
>/app/.node-gyp/10.0.0/include/node/node.h:88:42: note: in definition of
>macro ‘NODE_DEPRECATED’
>       __attribute__((deprecated(message))) declarator
>       ^
>   In file included from ../src/node-boilerplate/scrypt_common.cc:28:0:
>../../nan/nan.h: In function ‘v8::Local<v8::Value>
>Nan::MakeCallback(v8::Local<v8::Object>, v8::Local<v8::String>, int,
>v8::Local<v8::Value>*)’:
>../../nan/nan.h:850:62: warning: ‘v8::Local<v8::Value>
>node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>,
>v8::Local<v8::String>, int, v8::Local<v8::Value>*)’ is deprecated: Use
>MakeCallback(..., async_context) [-Wdeprecated-declarations]
>       v8::Isolate::GetCurrent(), target, symbol, argc, argv);
>       ^
>       In file included from ../../nan/nan.h:49:0,
>       from ../src/node-boilerplate/scrypt_common.cc:28:
>  /app/.node-gyp/10.0.0/include/node/node.h:165:50: note: declared here
>       NODE_EXTERN v8::Local<v8::Value> MakeCallback(
>       ^
>/app/.node-gyp/10.0.0/include/node/node.h:88:42: note: in definition of
>macro ‘NODE_DEPRECATED’
>       __attribute__((deprecated(message))) declarator
>       ^
>   In file included from ../src/node-boilerplate/scrypt_common.cc:28:0:
>../../nan/nan.h: In function ‘v8::Local<v8::Value>
>Nan::MakeCallback(v8::Local<v8::Object>, const char*, int,
>v8::Local<v8::Value>*)’:
>../../nan/nan.h:865:62: warning: ‘v8::Local<v8::Value>
>node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, const char*,
>int, v8::Local<v8::Value>*)’ is deprecated: Use MakeCallback(...,
>async_context) [-Wdeprecated-declarations]
>       v8::Isolate::GetCurrent(), target, method, argc, argv);
>       ^
>       In file included from ../../nan/nan.h:49:0,
>       from ../src/node-boilerplate/scrypt_common.cc:28:
>  /app/.node-gyp/10.0.0/include/node/node.h:158:50: note: declared here
>       NODE_EXTERN v8::Local<v8::Value> MakeCallback(
>       ^
>/app/.node-gyp/10.0.0/include/node/node.h:88:42: note: in definition of
>macro ‘NODE_DEPRECATED’
>       __attribute__((deprecated(message))) declarator
>       ^
>   In file included from ../src/node-boilerplate/scrypt_common.cc:28:0:
>../../nan/nan.h: In member function ‘v8::Local<v8::Value>
>Nan::Callback::Call_(v8::Isolate*, v8::Local<v8::Object>, int,
>v8::Local<v8::Value>*) const’:
>../../nan/nan.h:1479:5: warning: ‘v8::Local<v8::Value>
>node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>,
>v8::Local<v8::Function>, int, v8::Local<v8::Value>*)’ is deprecated:
>Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
>       ));
>       ^
>       In file included from ../../nan/nan.h:49:0,
>       from ../src/node-boilerplate/scrypt_common.cc:28:
>  /app/.node-gyp/10.0.0/include/node/node.h:172:50: note: declared here
>       NODE_EXTERN v8::Local<v8::Value> MakeCallback(
>       ^
>/app/.node-gyp/10.0.0/include/node/node.h:88:42: note: in definition of
>macro ‘NODE_DEPRECATED’
>       __attribute__((deprecated(message))) declarator
>       ^
>scrypt.target.mk:115: recipe for target
>'Release/obj.target/scrypt/src/node-boilerplate/scrypt_common.o' failed
>make: Leaving directory
>'/tmp/build_431d75b25ac1fa5430285940870b0ae7/node_modules/scrypt/build'
>make: ***
>[Release/obj.target/scrypt/src/node-boilerplate/scrypt_common.o] Error
>1
>       gyp ERR! build error
>       gyp ERR! stack Error: `make` failed with exit code: 2
>gyp ERR! stack     at ChildProcess.onExit
>(/tmp/build_431d75b25ac1fa5430285940870b0ae7/.heroku/node/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:258:23)
>       gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
>gyp ERR! stack     at Process.ChildProcess._handle.onexit
>(internal/child_process.js:225:12)
>       gyp ERR! System Linux 4.4.0-1014-aws
>gyp ERR! command
>"/tmp/build_431d75b25ac1fa5430285940870b0ae7/.heroku/node/bin/node"
>"/tmp/build_431d75b25ac1fa5430285940870b0ae7/.heroku/node/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js"
>"rebuild"
>gyp ERR! cwd
>/tmp/build_431d75b25ac1fa5430285940870b0ae7/node_modules/scrypt
>       gyp ERR! node -v v10.0.0
>       gyp ERR! node-gyp -v v3.6.2
>       gyp ERR! not ok
>info Visit https://yarnpkg.com/en/docs/cli/install for documentation
>about this command.
>-----> Build failed
>       
>We're sorry this build is failing! You can troubleshoot common issues
>here:
>     https://devcenter.heroku.com/articles/troubleshooting-node-deploys
>       
>       Some possible problems:
>       
>       - Dangerous semver range (>) in engines.node
>https://devcenter.heroku.com/articles/nodejs-support#specifying-a-node-js-version
>       
>       Love,
>       Heroku
>       
> !     Push rejected, failed to compile Node.js app.
> !     Push failed
>
>```

-- 
Signed
Sheogorath


Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by crcastle** [comment]
One fix is to more explicitly specify the versions of node hackmd will work with.  Right now it is `>=6.x`. Maybe change it to `>=6.x <10`?

https://github.com/hackmdio/hackmd/blob/master/package.json#L135

**Comment by SISheogorath** [comment]
Should be "fixed" in 1.1.1-ce
